src/lib/routing-tables.js

```diff
@@ -0,0 +1,219 @@
+          connector: this.baseURI,
+          points: combinedRoute.getPoints(),
+          min_message_window: combinedRoute.info.minMessageWindow,
+          source_account: this._getAccount(this.baseURI, sourceLedger)
+        })
+      }
+    })
+    return routes
+  }
+
+  _getAccount (connector, ledger) {
+    return this.accounts[connector + ';' + ledger]
+  }
+
+  _getLocalRoute (ledgerA, ledgerB) {
+    const routesFromAToB = this.sources[ledgerA].destinations.get(ledgerB)
+    if (!routesFromAToB) return
+    return routesFromAToB.get(this.baseURI)
+  }
+
+  /**
+   * Find the best intermediate ledger (`ledgerB`) to use after `ledgerA` on
+   * the way to `ledgerC`.
+   * This connector must have `[ledgerA, ledgerB]` as a pair.
+   *
+   * @param {URI} ledgerA
+   * @param {URI} ledgerC
+   * @param {String} finalAmount
+   * @returns {Object}
+   */
+  findBestHopForDestinationAmount (ledgerA, ledgerC, finalAmount) {
+    const nextHop = this._findBestHopForDestinationAmount(ledgerA, ledgerC, +finalAmount)
+    if (!nextHop) return
+    const ledgerB = nextHop.info.nextLedger
+    const routeFromAToB = this._getLocalRoute(ledgerA, ledgerB)
+    const isLocal = this.baseURI === nextHop.bestHop
+    return {
+      connector: nextHop.bestHop,
+      sourceLedger: ledgerA,
+      // Prevent 'BigNumber Error: new BigNumber() number type has more than 15 significant digits'
+      sourceAmount: nextHop.bestCost.toFixed(15),
+      destinationLedger: ledgerB,
+      destinationAmount: routeFromAToB.amountAt(nextHop.bestCost).toFixed(15),
+      destinationDebitAccount: this._getAccount(this.baseURI, ledgerB),
+      destinationCreditAccount: isLocal ? null : this._getAccount(nextHop.bestHop, ledgerB),
+      finalLedger: ledgerC,
+      finalAmount: finalAmount,
+      minMessageWindow: nextHop.info.minMessageWindow
+    }
+  }
+
+  /**
+   * @param {URI} ledgerA
+   * @param {URI} ledgerC
+   * @param {String} sourceAmount
+   * @returns {Object}
+   */
+  findBestHopForSourceAmount (ledgerA, ledgerC, sourceAmount) {
+    const nextHop = this._findBestHopForSourceAmount(ledgerA, ledgerC, +sourceAmount)
+    if (!nextHop) return
+    const ledgerB = nextHop.info.nextLedger
+    const routeFromAToB = this._getLocalRoute(ledgerA, ledgerB)
+    const isLocal = this.baseURI === nextHop.bestHop
+    return {
+      connector: nextHop.bestHop,
+      sourceLedger: ledgerA,
+      sourceAmount: sourceAmount,
+      destinationLedger: ledgerB,
+      destinationAmount: routeFromAToB.amountAt(sourceAmount).toFixed(15),
+      destinationDebitAccount: this._getAccount(this.baseURI, ledgerB),
+      destinationCreditAccount: isLocal ? null : this._getAccount(nextHop.bestHop, ledgerB),
+      finalLedger: ledgerC,
+      finalAmount: nextHop.bestValue.toFixed(15),
+      minMessageWindow: nextHop.info.minMessageWindow
+    }
+  }
+}
+
+;[
+  'findBestHopForSourceAmount',
+  'findBestHopForDestinationAmount'
+].forEach(function (fn) {
```

Comment:
This seems like a janky way of calling these two functions. Why not just separate this out into two calls?